Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -1767,7 +1773,14 @@ class ExprWalker : PythonWalker {
         internal async Task UnloadFileAsync(AnalysisEntry entry) {
             _analysisComplete = false;
 
-            entry?.TryGetBufferParser()?.ClearBuffers();
+            var bp = entry?.TryGetBufferParser();
+            if (bp != null) {
+                bp.ClearBuffers();
+            } else {
+                _services.MaybeErrorTaskProvider?.ClearErrorSource(entry.Path, ParserTaskMoniker);
+                _services.MaybeErrorTaskProvider?.ClearErrorSource(entry.Path, UnresolvedImportMoniker);
+                _services.MaybeCommentTaskProvider?.ClearErrorSource(entry.Path, ParserTaskMoniker);
```

Comment:
Do you need `_services.MaybeCommentTaskProvider?.ClearErrorSource(entry.Path, InvalidEncodingMoniker);` here as well?

Reply:
Probably... that one is fairly inconsistent right now...

Ideally it would be detected and reported as a regular error by the language server, and then we wouldn't need a separate moniker for it.